### PR TITLE
Implement more (all remaining?) trigonometric functions

### PR DESCRIPTION
**Pull request by jongleur1983**
#### What's this PR do?
implements more functions in the trigonometric area, in particular
COTH, CSC, ACOT, ACOTH, SEC, SECH

#### Where should the reviewer start?
To follow as easy as possible, it might be best to review commit by commit, as each commit implements one function and it's tests.

#### How should this be manually tested?
compare with how Excel behaves

- [ ] C# Code Review: @csreviewer
- [ ] Test Automation Review: @csreviewer

### CODE REVIEW

**Comment by jongleur1983** [comment]
I have no idea why the build fails here. Any help appreciated. Works for me at home and code looks fine IMHO (but I may be stupid of course)

File: ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs [hunk 1]
```diff
@@ -699,6 +744,19 @@ private static object RoundUp(List<Expression> p)
             return Math.Floor(value * Math.Pow(10, digits)) / Math.Pow(10, digits);
         }
 
+        private static object Sec(List<Expression> p)
+        {
+            if (double.TryParse(p[0], out double number))
```
Comment (igitur) [inline ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: Use C#6 syntax

Comment (jongleur1983) [reply ClosedXML/Excel/CalcEngine/Functions/MathTrig.cs]: Thanks. I'm quite used to C#7 already, didn't even think about using it that way.

**Comment by igitur** [comment]
You've signed up for your t-shirt, right?

**Comment by jongleur1983** [comment]
;) thanks, of course I have.

**Pull request merged**